File: Solution/1_EASY/1_3_TwoNumber.js
function twoNumberSum(array, targetSum) {
  let left = 0;
  let right = array.length - 1;
  array = array.sort((a, b) => a - b);
  while (left < right) {
    if (array[left] + array[right] == targetSum)
      return [array[left], array[right]];
    else if (array[left] + array[right] < targetSum) left++;
    else right--;
  }
  return [];
}

console.log(twoNumberSum([3, 5, -4, 8, 11, 1, -1, 6], 10));
console.log(twoNumberSum([3, 5, -4, 8, 11, 1, -1, 6], 11));
console.log(twoNumberSum([3, 5, -4, 8, 11, 1, -1, 6], 99));

File: Solution/1_3_TwoNumber.py
# Time Complexity will Be O(nlogn) Because sotring Space Complexit will be O(n)
def twoNumberSum(array, targetSum):
    array = sorted(array)
    left = 0
    right = len(array)-1
    while(left < right):
        if(array[left]+array[right] == targetSum):
            return [array[left], array[right]]
        elif(array[left]+array[right] > targetSum):
            right -= 1
        else:
            left += 1
    return []


print(twoNumberSum([3, 5, -4, 8, 11, 1, -1, 6], 10))
print(twoNumberSum([3, 5, -4, 8, 11, 1, -1, 6], 11))
print(twoNumberSum([3, 5, -4, 8, 11, 1, -1, 6], 99))
